Make installable on PHP 8.x

evs-xsarus opened the pull request:
New PHP subversions hardly ever break, remove the hard upper version constraint.

maksim-grib commented:
This PR is made as part of agreement with Bluebird Day.

It allows you to include catalog rule prices for non logged in users to the feed. This needs to be tested still.

edwinljacobs commented on src/Model/Write/Products/CollectionDecorator/Price.php:
```diff
@@ -61,13 +72,31 @@ public function decorate(Collection $collection)
             ]);
         $collectionQuery = $collectionSelect->query();
 
+        $catalogRulePrices = $this->getCatalogRulePrices($websiteId, $collection->getIds());
```
It would be faster to join the catalogrule_product_price table for example:

`$collectionSelect->joinLeft(
            ['rp' => $this->catalogRuleResource->getTable('catalogrule_product_price')],
            "rp.product_id = e.entity_id AND rp.website_id = $websiteId AND rp.customer_group_id = $groupId AND rp.rule_date = '$now'",
            ['rule_price' => 'rp.rule_price']
        );`

One can then skip `$row['rule_price'] = isset($catalogRulePrices[$entityId]) ? $catalogRulePrices[$entityId] : null;` and also the method `protected function getCatalogRulePrices($websiteId, array $productIds)` could be skipped

It could be that in the future the export mdoule gets refactored to make use of api methods but as is that is not the case.

We tested the changes as is and that seems to work.

maksim-grib replied:
Yes, I thought about that solution as well. It won't be faster, but it will use less memory in that case. I can refactoring it this way if you seeing it that way.

edwinljacobs replied:
Yeah I think that would be better (as it is less code) when done I will accept the pull request and tag it :)
Thanks for your work!

maksim-grib commented:
Hi there! I've tried to refactor code using approach we discussed above. Hope now it looks better, but please let me know if there is something else I need to change. And thanks for your input.

edwinljacobs commented:
Merged and tagged! Thanks for your work!

evs-xsarus committed:
Make installable on PHP 8.x

New PHP subversions hardly ever break, remove the hard upper version constraint.

ahuininga-orisha committed:
fix: make installable on php 8.3
